### DISCUSSION TITLE
[Feature] Add optional params to getRandom in API

### DISCUSSION THREAD

**Discussion by 3rob3** [immich-app/immich#6539] [Feature Request]
### The feature

It would be great if the API getRandom (/api/asset/random) accepted optional params for isFavorite and album.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Another option would be to update the [search assets endpoint](https://immich.app/docs/api/search-assets) with a random sort option.

**Reply by singletrackdev** [reply]
Adding a random sort and limit to the search assets endpoint would give significant flexibility, allowing for example a random video to be returned.

**Comment by jschwalbe** [comment]
Have been playing around with https://github.com/damongolding/immich-kiosk which ends up being a great way to display photos from immich. Bumping up against an issue with the random function (see [this discussion](https://github.com/damongolding/immich-kiosk/discussions/76)). Would be neat to have either an updated getRandom api function (with albumID and personID) or like @bo0tzz mentioned, an updated search endpoint (which seems to have changed since January as that link is dead).

Any further thoughts? Thanks!

**Reply by damongolding** [reply]
Beat me to it huh @jschwalbe 😉
